src/main/java/com/example/SS2_Backend/util/SampleDataGenerator.java

```diff
@@ -1,58 +1,71 @@
 package com.example.SS2_Backend.util;
 
-import com.example.SS2_Backend.model.stableMatching.Individual;
+import com.example.SS2_Backend.dto.mapper.StableMatchingProblemMapper;
+import com.example.SS2_Backend.dto.request.NewStableMatchingProblemDTO;
 import com.example.SS2_Backend.model.stableMatching.Matches.Matches;
-import com.example.SS2_Backend.model.stableMatching.StableMatchingProblem;
+import com.example.SS2_Backend.model.stableMatching.StableMatchingRBOProblem;
+import com.example.SS2_Backend.ss.smt.MatchingProblem;
 import lombok.Data;
 import lombok.extern.slf4j.Slf4j;
 import org.moeaframework.Executor;
 import org.moeaframework.core.NondominatedPopulation;
 import org.moeaframework.core.Solution;
 
-import java.util.ArrayList;
-import java.util.List;
+import java.util.Map;
 import java.util.Random;
 
+enum MATCHING_PROBLEM_TYPE {
+    MTM,
+    OTM,
+    OTO
+}
 /**
  * Stable Matching Problem Testing Space.
  */
 @Data
 @Slf4j
 public class SampleDataGenerator {
-
     // Configuration parameters
-    private String[] propNames;
-    private int numberOfSet1;
-    private int numberOfSet2;
-    private int set1Cap = 10; // Default capacity for set 1
-    private int set2Cap = 10; // Default capacity for set 2
-    private boolean randCapSet1 = false; // Randomize capacity for set 1
-    private boolean randCapSet2 = false; // Randomize capacity for set 2
-    private String f1 = "none";  // Evaluation function for set 1
-    private String f2 = "none";  // Evaluation function for set 2
-    private String fnf = "none"; // Fitness function
+    private MATCHING_PROBLEM_TYPE matchingProblemType = MATCHING_PROBLEM_TYPE.MTM;
+    // problemSize
+    private int individualNum;
+    private int numberOfProperties;
+    // max capacity tiêu chuẩn cho mỗi set dạng map<int, int>, vd: với MTM: {0: 2, 1: 10}, 3Set: {0: 1, 1: 10, 2: 12}
+     private int numberOfSet1;
```

Comment:
private int numberOfSet1;
     private int numberOfSet2;
     
     
     
     sửa lại thành mảng để chứa các number of set

Reply:
Done